File: 14-Longest-Common-Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        pre = \\
        sortedStrs = sorted(strs, key=str.lower)
        smallest = sortedStrs[0]
        longest = sortedStrs[-1]

        for i in range(len(smallest)):
            if smallest[i] == longest[i]:
                pre += smallest[i]
            else:
                return pre

        return pre
        

File: 14-Longest-Common-Prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
    if (!strs.length) return ''
    strs = strs.sort((a, b) => a.length - b.length)
    let pref = ''

    for (let i = 0; i < strs[0].length; i++) {
        let c = strs[0][i]
        if (!c) return ''

        if (strs.every(s => s[i] === c)) pref += c
        else break
    }

    return pref
};